### PR TITLE
[Cleaner] Ensure Obfuscation of Domains in SSSD conf files

### PR DESCRIPTION
**Pull request by bryanfraschetti**
Signed-off-by: Bryan Fraschetti \<bryan.fraschetti@canonical.com\>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [x] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

**Comment by bryanfraschetti** [comment]
Hmm. I've stumbled on something strange here
I was developing the tests locally and all were passing except `test_hostname_prepper_sssd_conf_values_in_items`. After mocking `host_prepper._get_conf_files` since the files aren't actually on disk, all were passing. In CI `test_hostname_prepper_sssd_fqdn_fully_obfuscated` failed so I cleared `.tox` and `/var/tmp/sos_avocado_testing/` and now I reproduce the failure locally so there must some weird caching of either test results or host/domain names

File: sos/cleaner/preppers/hostname.py [hunk 1]
```diff
@@ -25,6 +28,85 @@ class HostnamePrepper(SoSPrepper):
...
+            return None
+        # Handle URI-style values, e.g. ldap://host.example.com:389
+        if '://' in value:
+            value = value.split('://', 1)[1]
+        value = value.split('/')[0]
+        if value.startswith('['):
+            value = value.strip('[]')
+        elif ':' in value:
+            value = value.split(':', 1)[0]
+        return value.strip()
```
Comment (pmoravec) [inline sos/cleaner/preppers/hostname.py]: Can't `urlparse(value).hostname` help here? Dont reinvent wheel until you need some specific one (which might be the case here..?)

File: sos/cleaner/preppers/hostname.py [hunk 2]
```diff
@@ -25,6 +28,85 @@ class HostnamePrepper(SoSPrepper):
...
+        items = []
+
+        paths = self._get_conf_files(archive)
+
+        for path in sorted(paths):
+            content = archive.get_file_content(path)
+            if not content:
+                continue
+            for line in content.splitlines():
+                line = line.split('#', 1)[0].split(';', 1)[0].strip()
```
Comment (pmoravec) [inline sos/cleaner/preppers/hostname.py:92]: maybe an explanation or an example of this split/strip helps..?

What if a user comments out whole line with a valid hostname? Technically the semantics of the config file is "no such (commented) hostname ever exists". Practically, the comment can be auxiliary and the hostname can be very valid (occurring in past logs, e.g.).

Not sure if my comment about `^#..` lines is much valid, or if it is worth adding complexity here for such a corner case (well, maybe simple `while line starts with '#', strip the char` can be prepped..?).

**Comment by pmoravec** [comment]
Ack up to the few comments. Ack to your easter egg / name in tests :)

File: sos/cleaner/preppers/hostname.py (hunk 1 above)
Comment (bryanfraschetti) [reply sos/cleaner/preppers/hostname.py]: I actually don't need this function, so I'll remove it

File: sos/cleaner/preppers/hostname.py (hunk 2 above)
Comment (bryanfraschetti) [reply sos/cleaner/preppers/hostname.py:92]: That's true, I was treating comments as not having any sensitive content, but there may be previous entries that have temporarily been commented out or left there for legacy's sake. I think `while line starts with '#', strip the char` makes sense

I think it is less likely that inline comments will have any personally identifiable information and is more likely to be unstructured text. I'll update the PR so that we don't ignore lines that start with a comment

**Review by pmoravec — APPROVED** [review]

**Review by TurboTurtle — APPROVED** [review]
